src/inputs/pipe.c

```diff
@@ -302,6 +304,51 @@ parse_progress(struct input_metadata *m, char *progress)
   m->song_length = (end - start) * 10 / 441; // Convert to ms based on 44100
 }
 
+static void
+parse_volume(char *volume)
+{
+  char *value_begin;
+  char *value_next;
+  char *value_end;
+  float airplay_volume;
+
+  if (!(value_begin = strtok_r(volume, ",", &value_next)))
+    return;
+
+  airplay_volume = strtof(value_begin, &value_end);
+
+  if ((errno == ERANGE) || (value_begin == value_end))
```

Comment:
Sorry, but you still have a problem here. You need to set errno = 0 before strtof, because if the conversion is successful strtof will not set errno, and who knows what the previous value might be.

You could also just remove all this checking and just concentrate on the value of airplay_volume:
-144: OK
between -30 - 0: OK
everything else: invalid